### PR TITLE
CAS exports allow to download the report for a certain month at 15th …

### PR DESCRIPTION
**Pull request by dmydlo**
…day of the next month
Hi @calellowitz @Rohit25negi,
I have changed the filter limitations for CAS exports. Right now you will be able to download the report for eg March starting from 15th April.
First change addresses the chosen month after getting on the page. Second available months to choose from after getting on the page. And the third one months available after changing year.
These changes are related to JIRA [ICDS-464](https://dimagi-dev.atlassian.net/browse/ICDS-464) ticket.
Need QA: No
Environment: n/a
Locally Tested: Yes
Regards, Dawid

### CODE REVIEW

File: custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js [hunk 1]
```diff
@@ -74,6 +76,8 @@ function CasExportController($window, $location, locationHierarchy, locationsSer
 
     vm.onSelectYear = function (year) {
         var latest = new Date();
+        var offset = latest.getDate() < 15 ? 2 : 1;
+        latest.setMonth(latest.getMonth() - offset);
```
Comment (calellowitz) [inline custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js]: what happens if `latest.getMonth()` returns a number less than the offset. For example in before 15 January

**Review by calellowitz — CHANGES_REQUESTED** [review]
You will also need to add these to the backend views, both for better assurance in case someone messes with the JS, and because these reports are accessible via an API and we need to limit that as well.

File: custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js (hunk 1 above)
Comment (dmydlo) [reply custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js]: It's going to change the year.

**Comment by calellowitz** [comment]
@dmydlo 
Any update here?

**Comment by dmydlo** [comment]
@calellowitz I have added the new date limits to the backend views.

**Review by calellowitz — APPROVED** [review]

**Pull request merged**